### ISSUE TITLE
Multiple Currencies

### ISSUE THREAD

**Issue by jakob-fuchs** [concrete5-community-store/community_store#106]
From looking at the provided classes it doesn't seem like multiple currencies are supported. There's also no mention of it in the roadmap page in the wiki.

Is that something that could be added in the future?

**Comment by Mesuva** [comment]
Well I have no objection to it being it added in the future, the sky is the limit!

But for something like this I personally struggle a bit as I've not had any experience with managing multi currency shops. I could imagine something where multiple currencies are entered into the dashboard, and then additional price fields appear through the product editing interface.

But what I don't really know is how complex this would need to get, would it need some sort of currency selector?  What happens when a currency isn't filled in for a product?  Does it then use multiple gateways, or would it just be showing an estimate based on the exchange rate?  

<head explodes>

**Comment by jakob-fuchs** [comment]
Yeah that's pretty complex stuff, hence I'd be looking for a premade solution. You could probably have periodically updated exchange rates and do it this way. No idea what the requirements for the different payment gateways are.

If you consider that products could have different prices for no reason depending on the country (see Adobe products) it gets even more complicated.

**Comment by appliculture** [comment]
Talking about multiple currencies is very complex I think if you want to cover it all.

From what I've seen in other products you can either define price per currencies per product or also use some finance API to get the price converted and rounded automatically.
The price conversion could be live or with an action button " Get price converted" to keep control in case of currency dive.
In addition some people would like to have a higher (or lower) price if from a different country so you have to be able to set a price ratio (x Euro = x USD converted * 1.5 ratio)

And this has to be set in general setting with ability to override with per product setting.

Pretty complex.

**Comment by Mesuva** [comment]
A few years back on a 5.6 ecommerce site I added live rate conversion - https://www.hobbytech.com.au/products/rc-helicopters/electric-helicopter/

All this does is go off and fetch some live-ish currency rates and do an on-page conversion with javascript of the prices it finds. It actually works pretty well in my opinion, but all it's doing is giving you an indication of the price you'll pay in your own currency, it's display online - the order and transaction are all in the base currency.

This approach would work with Community Store though, if we just considered the display only of prices, not the rest of the complexity.

**Comment by Remo** [comment]
I've had a lot of discussions about this in the past and my conculsion is clear: Don't do this!

* It's not a problem to pay in a different currency in pretty much all cases. My account is in CHF, I pay in EUR, so what.
* An indicator is nice and helpful, in the end you'll mostly see what you're actually going to pay too. Paypal does that for example.
* Automatically calculated prices are ugly. Not long and you'll get a call from the director of sales telling you that a price of €40.02 isn't acceptable. You add some magic rounding algorithm that results in €39.95. The next day you're selling an article for $0.01, you're algorithm rounds this to CHF 0.05 and guess what, you're going to get another call because one customer wants to order 1000 pieces and the price went up like crazy.

You fix that problem and eventually you're going to get rid of multiple currencies because it just doesn't work well and only causes problems.

**Comment by Mesuva** [comment]
@remo That all makes sense to me. So you're suggesting that some little tool or link on a website to make it convenient for a customer to do a currency conversion _just for reference_ isn't a bad idea, but trying to do some magic to make the prices change across the site itself is potentially asking for trouble.

In my example there is a ~ symbol when the prices are calculated, but I can easily see how it could be missed or simply give the wrong impression. Once you get through to the checkout the prices are no longer converted and it shows in AUD for the actual transaction. It's not actually changing prices, just do the conversions on screen. 

I buy stuff in US dollars reasonably often. I just do a quick currency conversion myself to know what I'll pay in Dollarydoos. I've never not paid for something I've needed/wanted because I couldn't pay in my own currency.

**Comment by Remo** [comment]
@Mesuva I understand that different people have different views and when a customer wants something you something have to do things you feel are wrong, but I always managed to get away with this and still keep the customer happy. Yes, I would recommend what you're describing and not more.

People buy things in China directly these days. Should those guys have 50 currencies on their site just because they ship their stuff to all those countries? No, it's fine, I pay you in USD and we are good to go. I can even type `100 USD in CHF` in google and get the price in my currency, admittedly something for nerds, but that's why we can put that on the site as an indicator.

**Comment by Mesuva** [comment]
I appreciate your thoughts on this Remo, it's just not something I've had much experience with. 100% agree with your thinking.

I reckon I do currency conversion myself on a daily basic, it even works in Spotlight Search on macs.
So yep, let's keep it simple.

**Comment by jasteele12** [comment]
I attempted to reply to this by email, but it appears that didn't work.  Here's my response to @Remo original response:
-----

I'm with Remo on this one.  Another important point here is tariffs.  Sell leather to someone in Italy from the U.S. and find out how bad the whole idea is (who would thinks shoes would be such a big deal, but they have different sizes too).  The rules change too often.

Best I've found is to give pointers/tools to the consumer on conversion rates, links to country customs/rules and make it clear, at best, it's just an estimate of final cost.

I don't believe it's a problem that can be solved (nor attempted to be addressed) with code.

Sorry, but it's just messy,
  John
-----
P.S. I'm guessing it's probably easier for China to ship than vice-versa
P.S.S. Don't get me started on Futures Exchanges and rates ;)
